src/scenes/AcceleratorRouter/People/index.tsx

```diff
@@ -11,7 +11,14 @@ import SingleView from './SingleView';
 const PeopleRouter = () => {
   return (
     <Routes>
-      <Route path={'new'} element={<NewView />} />
+      <Route
+        path={'new'}
+        element={
+          <PersonProvider>
+            <NewView />
+          </PersonProvider>
```

Comment:
Not sure if having the same provider in two different places in a tree of routes is considered bad practice. I tried moving it to the top level but then it isn't able to get the user ID from the path on the edit and single views.

Reply:
It isn't ideal, because the provider will be instantiated twice, which negates the benefit of a shared provider.

I think in this case you don't need the provider to be wrapped around the `NewView`. It seems the only functionality being passed into that view is the ability to set the user ID and the `update` function. Since the `update` functionality doesn't really provide any meaningful state that should be shared, I think it should just be thrown into a custom hook if you want to reduce the duplicate code. See [this hook](https://github.com/terraware/terraware-web/blob/main/src/components/DeliverableView/useUpdateDeliverable.ts).

The idea is that the state related to the request flow is encapsulated in the hook, which makes it easy to interact with the APIs without putting a bunch of boilerplate in separate views/components. Since the hook will be instantiated in multiple places, they are also not "shared", but this is OK because there isn't anything that needs to be shared.

Reply:
> It seems the only functionality being passed into that view is the ability to set the user ID and the update function.

Setting the user ID triggers the provider to fetch that user's details which are then passed back down to `NewView` to show in the form. How would I make that work with a hook?

Alternately, I could maybe grab the user ID from the URL in `EditView` and `SingleView` instead of in the provider, and then call the provider's `setUserId` to trigger loading the user's data. Then the whole tree of routes can be wrapped in a single provider rather than needing two of them. Does that seem like a sane approach?

Reply:
I am going to give that a try just to see what it looks like (but happy to throw it out if it's not the right thing to do).

Reply:
I see. I think that makes sense the way you have it then. I may be over complicating this, since this is already a bit complicated with the fact that we have a form that is eventually pre-populated with some remote data.

It seems that you do in fact need a way to have the consumer set the user ID within the provider in the event that it isn't available in the URL.